v3/bind.go

```diff
@@ -733,3 +733,115 @@ RESP:
 
 	return nil, GetLDAPError(packet)
 }
+
+// ServerBindStep sends the SASLBindRequest and return the result code, servercred and errors.
+// If the result code is not SUCCESS(0x00) or IN_PROGRESS(0x0e), it's also considered as an error.
+func (l *Conn) ServerBindStep(clientCred []byte, dn string, methodName string, controls []Control) (resultCode uint16, serverCred []byte, err error) {
+	packet := ber.Encode(ber.ClassUniversal, ber.TypeConstructed, ber.TagSequence, nil, "LDAP Request")
```

Comment:
Most of this (until line 770) is a duplication of what `doRequest` does. The function signature also mismatches from the style how other functions like `SimpleBind` are structure. They take in a struct (which also satisfies the `request` interface). Additionally, there are simple wrappers for those functions like `conn.Bind` for simple use.

In my opinion we should stick to one style here and not mix them up. Could you please adapt your two functions please?